doc/interceptors.md

```diff
@@ -0,0 +1,61 @@
+## HTTP Interceptors
+
+With the release of sync-android version TBD a new HTTP Interceptor API was introduced.
+HTTP Interceptors allow the developer to modify the HTTP Requests and Responses during
+the replication process.
+
+In order to make changes to a HTTP request, you need to implement the interface
+`HTTPConnectionRequestInterceptor`, for HTTP responses, you implement the interface
+`HTTPConnectionResponseInterceptor`. A Class may implement both interface without causing
+problems, an example of this is the `CookieInterceptor` class.
+
+In order to add an Http Interceptor to a replication, you call the `addResponseInterceptors` or
+`addRequestInterceptors` on the `ReplicatorBuilder` class.
+
+For example, this is how you would add an instance of `CookieFilter` to a pull replication.
+
+```java
+import com.cloudant.sync.replication.Replicator
+import com.cloudant.sync.replication.ReplicatorBuilder
+import com.cloudant.sync.datastore.Datastore
+import com.cloudant.sync.http.CookieFilter
+
+
+Datastore ds =  manager.open("my_datastore");
+URI cloudant = new URI("https://username.cloudant.com");
+CookieInterceptor interceptor = new CookieInterceptor("username","password");
+
+
+
+Replicator replicator = ReplicatorBuilder.pull()
+              .from(cloudant)
+              .to(ds)
+              .addRequestInterceptors(interceptor)
+              .addResponseInterceptors(interceptor)
+              .build();
+
+replicator.start();
+
+```
+
+Interceptors can be used to implement your own authentication schemes, for example OAuth, or
+provide custom headers so you can perform your own analysis on usage. For an example
+of how to implement a filter, see  the `CookieInterceptor class.
+
+
+## Things Not to do
+
+The `HttpConnectionInterceptorContext` object provides access to the underlying
+`HttpConnection` and `HttpURlConnection` classes. This allows you to change
+settings and interact with the connection in ways would could potentially cause
+errors in the replicator.
+
+For example, reading the `HttpURlConnection`'s `InputStream` will cause the
```

Comment:
Avoid implying a class name is an instance of itself, which you do twice in the above, instead:

"reading a `HTTPURLConnection` object's input stream will"